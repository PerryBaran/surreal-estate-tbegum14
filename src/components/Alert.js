import React from "react";
import "../styles/alert.css"

export default function Alert({message, success}){
    if(!message){
        return null
    }
    return (<div className={`alert alert-${success? "success":"error"}`} >
        <p>{message}</p>
    </div>)
}