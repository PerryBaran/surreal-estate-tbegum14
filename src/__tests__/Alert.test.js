import React from "react";
import {render} from "@testing-library/react"
import Alert from "../components/Alert"

describe("Alert", ()=>{
    it("renders correctly", ()=>{
        const {asFragment} = render(<Alert />)
        expect(asFragment).toMatchSnapshot()
    }),

    it("displays an error message", ()=>{
        const {getByText} = render(<Alert message="Error!" />)

        expect(getByText(/Error/).textContent).toBe("Error!")
    }),

    it("displays a success message", ()=>{
        const {getByText} = render(<Alert message="Success!" success />)

        expect(getByText(/Success/).textContent).toBe("Success!")
    }),

    it("does not render when message is null", ()=>{
        const {asFragment} = render(<Alert message="" />)

        expect(asFragment).toMatchSnapshot()
    })
})