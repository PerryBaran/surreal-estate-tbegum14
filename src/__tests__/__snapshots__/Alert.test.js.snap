// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert does not render when message is null 1`] = `[Function]`;

exports[`Alert renders correctly 1`] = `[Function]`;
